ligo/stablecoin/operator.ligo

```diff
@@ -0,0 +1,180 @@
+// SPDX-FileCopyrightText: 2020 tqtezos
+// SPDX-License-Identifier: MIT
+
+(*
+ * Operator library for stablecoin smart-contract
+ *)
+
+#include "hooks.ligo"
+
+type owner_tokens is map (address, set (token_id))
```

Comment:
Why do we need such a type if we support only one `token_id`? Can't it be simplified?

Reply:
I thought that's the part of specification as well 🤔

Reply:
It's not part of the specification, apparently this type is only used in `validate_operators` and you put `transfer_operation.1.1.0` into its `set (token_id)` which must be `0n`. So this set always contains one item which is `0n`.

Reply:
Then you pass it to `is_operator_impl` which does `validate_operator_token_type`, but token type is also validated in `transfer_tokens`, so you validate twice.